features/auth/useAuthStore.ts

```diff
@@ -49,6 +50,7 @@ export default function useAuthStore() {
     "auth.userId",
     null
   );
+  const [token, setToken] = usePersistedState<string | null>("auth.token", null)
```

Comment:
Storing token in local storage is not a great idea according to the [OWASP security guidelines](https://cheatsheetseries.owasp.org/cheatsheets/HTML5_Security_Cheat_Sheet.html#local-storage). We should be implementing a similar HttpOnly session cookies mechanism like in the old API instead with a CSRF token.

Figuring out whether someone is authenticated then becomes a case of making a request to an authenticated endpoint, and if it's successful and not returning a 401, then you can set the user as authenticated. I suspect this will need the backend to update first to not treat the token as a session as that's not what it's designed for.